src/state/application/hooks.ts

```diff
@@ -0,0 +1,11 @@
+import { useSelector } from 'react-redux'
+import { useWeb3React } from '@web3-react/core'
+import { AppState } from '../index'
+
+export function useBlockNumber(): number | undefined {
```

Comment:
Could our existing 'current block' implementation could be used instead? https://github.com/pancakeswap/pancake-frontend/blob/7d10afe553d5c8ef6a3b1351c176de397607c249/src/state/hooks.ts#L356.
 
(dispatched by)

https://github.com/pancakeswap/pancake-frontend/blob/7d10afe553d5c8ef6a3b1351c176de397607c249/src/state/hooks.ts#L85

If not, do we refactor other bits of the app to use this block value?

As the `state/application` folder seems to entirely be storing current block state, maybe it should be renamed?